From b09977763179805efc07b1f8648309d44455d5d7 Mon Sep 17 00:00:00 2001
From: ksrt12 <kazakov12stepan2012@live.ru>
Date: Fri, 28 Jul 2017 19:51:06 +0300
Subject: [PATCH 2/2] 2

---
 tools/changelog.sh | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/tools/changelog.sh b/tools/changelog.sh
index 7efceae3..c6c8add4 100755
--- a/tools/changelog.sh
+++ b/tools/changelog.sh
@@ -31,6 +31,8 @@ changelog_path_name=CHANGELOG.mkdn #changelog file path/name.extension
 source_name="Resurrection Remix Nougat" #Name to display in changelog.md top before version
 # input variables end
 
+if [ "$dtl" != "0" ];then
+
 export Changelog=$source_tree/$changelog_path_name
 export Temp_Changelog=$source_tree/$changelog_path_name.temp
 
@@ -87,5 +89,5 @@ then
 	echo "$(cat $Temp_Changelog)\n$(cat $Changelog)" > $Changelog
 	rm -f $Temp_Changelog
 fi
-
+fi
 echo -e ${ylw}"\n √ Changelog successfully generated.\n"${txtrst}
-- 
2.13.0

